### PR TITLE
Add signup endpoint

### PR DESCRIPTION
**Pull request by olatomide007**
Confirmed Password thrown away.

### CODE REVIEW

File: app/api/router.py [hunk 1]
```diff
@@ -2,9 +2,18 @@
 
 from api.example.views import router as example_router
 from api.system.views import router as system_router
-from api.search.views import router as search_router
+ master
```
Comment (a11rew) [inline app/api/router.py]: Are these 'master' lines artifacts from a merge or did you mean to add them? Please remove them if you did not mean to. If you did mean to, why are they necessary?

File: app/api/signup/schemas.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# from datetime import datetime
+# from typing import Optional
+# from uuid import UUID
+
+from pydantic import BaseModel
+
+
+class Signup(BaseModel):
+    # id: Optional[UUID] = None
```
Comment (a11rew) [inline app/api/signup/schemas.py]: Please *delete* - not comment out lines you do not use. Only leave comments if they're actual code comments

File: app/api/signup/views.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from fastapi import APIRouter
+from .schemas import Signup
+
+router = APIRouter()
+
+@router.get("/", response_model= Signup)
```
Comment (a11rew) [inline app/api/signup/views.py]: Signup should be a POST endpoint with the `first_name` etc included in the request body.

File: app/api/signup/views.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+async def signup_user(first_name: str,last_name: int, email: str, password: str):
+                    show = {
```
Comment (a11rew) [inline app/api/signup/views.py]: Please use a variable descriptive of what this does. `resp` or `response` works way better here than a seemingly random `show`

**Review by a11rew — CHANGES_REQUESTED** [review]

**Comment by a11rew** [comment]
You have merge conflicts to resolve.

**Comment by olatomide007** [comment]
Suggested Changes have been made

File: app/api/signup/views.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+from fastapi import APIRouter
+from .schemas import Signup
+
+
+router = APIRouter()
+
+
+@router.post("/", response_model= Signup)
+async def signup_user(first_name: str,last_name: int, email: str, password: str):
```
Comment (a11rew) [inline app/api/signup/views.py]: Defining the parameters here makes them query parameters, they need to be in the request body.

See https://fastapi.tiangolo.com/tutorial/body/#declare-it-as-a-parameter

**Review by a11rew — CHANGES_REQUESTED** [review]
Also, don't delete the example and system API routers, others will need them for reference

File: app/api/signup/views.py (hunk 5 above)
Comment (olatomide007) [reply app/api/signup/views.py]: Okay. I have done that:  Making parameters in the request body and example and system API routers restored.

**Pull request merged**